src/Build/BackEnd/BuildManager/BuildSubmission.cs

```diff
@@ -156,7 +156,7 @@ internal void CompleteResults(BuildResult result)
             // this one.)
             ErrorUtilities.VerifyThrow(result.ConfigurationId == BuildRequest.ConfigurationId, "BuildResult doesn't match BuildRequest configuration");
 
-            if (BuildResult == null)
+            if (BuildResult is null || (BuildResult.OverallResult == BuildResultCode.Success && result is not null))
```

Comment:
I looked at this again more closely, and I think my change may have been a bit too simplistic. Tell me if this makes sense.

When we were completing a particular result, we set BuildResult to be that result if BuildResult was null (i.e., this is the first result) but did not change after that. When we then checked for completion, we used BuildResult. That means that after the first BuildResult, we didn't care about future BuildResults. I think this worked well enough because it was unusual for users to submit several builds for a single configuration by themselves and more unusual to have some (but not the first) fail.

The new version I have here goes back to setting BuildResult based on the first result, but when checking for completion, it passes in each result as it comes along, ensuring each result has an opportunity to set, for example, BuildResult.OverallResult.

I think that is more understandable and doesn't need a comment, but feel free to disagree, and I can add one.